File: src/main/java/com/freya02/botcommands/api/prefixed/annotations/ID.java
package com.freya02.botcommands.api.prefixed.annotations;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Use this annotation to specify this int/long parameter is supposed to be the ID of an entity (message, user, guild, channel, role...)
 */
@Retention(RetentionPolicy.RUNTIME)
@Target({ElementType.PARAMETER})
public @interface ID {

}

File: src/main/kotlin/com/freya02/botcommands/api/commands/prefixed/annotations/ID.kt
package com.freya02.botcommands.api.commands.prefixed.annotations

/**
 * Adapts the help content of this parameter as an ID of an entity (message, user, guild, channel, role...)
 */
@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.VALUE_PARAMETER)
annotation class ID
